contract/models/contract.py

```diff
@@ -346,7 +346,7 @@ def recurring_create_invoice(self):
             body = Markup(
                 self.env._("Contract manually invoiced: %(invoice_link)s")
             ) % {"invoice_link": invoice._get_html_link(title=invoice.name)}
-            self.message_post(body=body)
+            invoice.message_post(body=body)
```

Comment:
The idea here is to log in the contract whenever the invoicing action is triggered manuall

```suggestion
            for contract in invoice.invoice_line_ids.contract_line_id.contract_id:
                contract.message_post(body=body)
```